test/integration/belongsToMany.test.js

```diff
@@ -101,5 +102,44 @@ describe('belongsTo', function () {
 
       expect(this.User.findAll, 'was called twice');
     });
+
+    it('batches to multiple findAll call with where + limit', async function () {
+      let members1 = this.project1.getMembers({ where: { awesome: true }, limit: 1 })
+        , members2 = this.project2.getMembers({ where: { awesome: true }, limit: 1 })
+        , members3 = this.project3.getMembers({ where: { awesome: true }, limit: 2 });
```

Comment:
Not currently testing where actually, add a few more projects and make one or two false with the same limit as one with true